hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -627,21 +614,20 @@ public void sendDatanodeCommand(SCMCommand<?> command,
    * @param target The datanode which will receive the command.
    * @param scmDeadlineEpochMs The epoch time in ms, after which the command
    *                           will be discarded from the SCMPendingOps table.
-   * @param datanodeDeadlineEpochMs The epoch time in ms, after which the
-   *                                command will be discarded on the datanode if
-   *                                it has not been processed.
    * @throws NotLeaderException
    */
   public void sendDatanodeCommand(SCMCommand<?> command,
       ContainerInfo containerInfo, DatanodeDetails target,
-      long scmDeadlineEpochMs, long datanodeDeadlineEpochMs)
+      long scmDeadlineEpochMs)
       throws NotLeaderException {
+    long datanodeDeadline =
+        scmDeadlineEpochMs - rmConf.getDatanodeTimeoutOffset();
```

Comment:
We could have a weird edge case where `datanodeDeadline` is negative. Let's have some sort of a run time validation?

Reply:
I don't think negative deadline could be a problem.  Offset would have to be 53+ years (and counting) for deadline to be negative.

However, validation is a good idea.  It can be done in `ReplicationManagerConfig#validate`, since `scmDeadlineEpochMs` is also calculated based on a config property.